tests/model/test_model.py

```diff
@@ -1360,6 +1361,139 @@ def test_can_user_edit_topic(
+                None,
+                id="topic_resolved:Zulip2.1+:ZFL0",
+            ),
+            case(
+                RESOLVED_TOPIC_PREFIX + "hi!",
+                11662271397,
+                0,
+                None,
+                None,
+                "hi!",
+                None,
+                id="topic_unresolved:Zulip2.1+:ZFL0",
+            ),
+            case(
+                RESOLVED_TOPIC_PREFIX + "hi!",
+                11662271397,
+                10,
+                86400,
+                None,
+                "hi!",
+                None,
+                id="topic_unresolved:Zulip2.1+:ZFL10",
+            ),
+            case(
+                "hi!",
+                11662271397,
+                162,
+                None,
+                86400,
+                RESOLVED_TOPIC_PREFIX + "hi!",
+                None,
+                id="topic_resolved:Zulip7.0+:ZFL162",
+            ),
+            case(
+                RESOLVED_TOPIC_PREFIX + "hi!",
+                11662271397,
+                162,
+                None,
+                259200,
+                "hi!",
+                None,
+                id="topic_unresolved:Zulip7.0+:ZFL162",
+            ),
+            case(
+                "hi!",
+                0,
+                12,
+                86400,
+                None,
+                "hi!",
+                " Time limit for editing topic has been exceeded.",
+                id="time_limit_exceeded:Zulip2.1+:ZFL12",
+            ),
+            case(
+                "hi!",
+                0,
+                162,
+                None,
+                259200,
+                "hi!",
+                " Time limit for editing topic has been exceeded.",
+                id="time_limit_exceeded:Zulip7.0+:ZFL162",
+            ),
+        ],
+    )
+    def test_toggle_topic_resolve_status(
+        self,
+        mocker,
+        model,
+        initial_data,
+        topic_name,
+        timestamp,
+        server_feature_level,
+        topic_editing_limit_seconds,
+        move_messages_within_stream_limit_seconds,
+        expected_new_topic_name,
+        expected_footer_error,
+        stream_id=1,
+    ):
+        model.initial_data = initial_data
+        model.server_feature_level = server_feature_level
+        initial_data[
+            "realm_community_topic_editing_limit_seconds"
+        ] = topic_editing_limit_seconds
+        initial_data[
+            "realm_move_messages_within_stream_limit_seconds"
+        ] = move_messages_within_stream_limit_seconds
+        # If user can't edit topic, topic (un)resolve is disabled. Therefore,
+        # default return_value=True
+        model.can_user_edit_topic = mocker.Mock(return_value=True)
+        model.get_latest_message_in_topic = mocker.Mock(
+            return_value={
+                "subject": topic_name,
+                "timestamp": timestamp,
+                "id": 1,
+            }
+        )
```

Comment:
Minor nits: if you have arbitrary constants, it's often useful to have them different, in case they 'work' when they are accidentally used in place of one another.

It's debatable whether a message id of 1 makes sense here, but it's a parameter so easily changed or parametrized later.